model/map-jpa/src/main/resources/META-INF/authz/resource/jpa-authz-resource-changelog-1.xml

```diff
@@ -129,4 +133,27 @@ limitations under the License.
         </createIndex>
     </changeSet>
 
+    <changeSet author="keycloak" id="authz-resource-15699-drop-constraint" dbms="postgresql">
+        <preConditions onSqlOutput="TEST" onFail="MARK_RAN">
+            <changeSetExecuted author="keycloak" id="authz-resource-2" changeLogFile="META-INF/authz/resource/jpa-authz-resource-changelog-1.xml" />
+        </preConditions>
+        <dropUniqueConstraint tableName="kc_authz_resource_attribute" constraintName="kc_authz_resource_attribute_fk_root_name_value_key" />
+    </changeSet>
+
+    <changeSet author="keycloak" id="authz-resource-15699-add-hash-column" dbms="postgresql">
+        <addColumn tableName="kc_authz_resource_attribute">
+            <column name="value_hash" type="VARCHAR(255)"/>
+        </addColumn>
+        <modifySql>
+            <append value="GENERATED ALWAYS AS (sha256(value::bytea)) STORED"/>
+        </modifySql>
+    </changeSet>
```

Comment:
We might think about reworking our custom `GeneratedColumnChange` to allow generation of columns that are not only based on json fields. It currently generates columns using only fields from json columns, but it shouldn't be hard to adapt it to make the above work as well. Something like 

```
<ext:addGeneratedColumn tableName="abc">
    <column name="value_hash" type="VARCHAR(255)" function="sha256(value::bytea)"  stored="true"/>
</ext:addGeneratedColumn>
```

But I would say this would be a nice customization for the future. For now it is fine as is.

Reply:
I agree it might be better to adapt the extension to allow for adding generated column not only for json. I'll try to look into it within this issue as follow-up task would complicate changelogs even more.